src/main/java/com/sleekbyte/tailor/Tailor.java

```diff
@@ -123,16 +146,85 @@ public static CommonTokenStream getTokenStream(File inputFile) throws IOExceptio
      *
      * @param tokenStream Token stream generated by lexer
      * @return Parse Tree or null if parsing error occurs (and debug flag is set)
+     */
+    public static Optional<TopLevelContext> getParseTree(Optional<CommonTokenStream> tokenStream) {
+        Optional<TopLevelContext> tree = Optional.empty();
+        if (!tokenStream.isPresent()) {
+            return tree;
+        }
+        SwiftParser swiftParser = new SwiftParser(tokenStream.get());
+        try {
+            if (!configuration.debugFlagSet()) {
+                swiftParser.removeErrorListeners();
+                swiftParser.addErrorListener(new ErrorListener());
+            }
+            tree = Optional.of(swiftParser.topLevel());
+        } catch (CliArgumentParserException e) {
+            handleCliException(e);
+        }
+        return tree;
+    }
+
+    /**
+     * Walks the provided parse tree using the list of listeners.
+     *
+     * @param listeners List of parse tree listeners.
+     * @param tree Parse tree.
+     */
+    public static void walkParseTree(List<SwiftBaseListener> listeners, TopLevelContext tree) {
+        ParseTreeWalker walker = new ParseTreeWalker();
+        for (SwiftBaseListener listener : listeners) {
+            // The following listeners are used by DeclarationListener to walk the tree
+            if (listener instanceof ConstantNamingListener || listener instanceof KPrefixListener) {
+                continue;
+            }
+            walker.walk(listener, tree);
+        }
+    }
+
+    /**
+     * Analyzes an individual file by creating the corresponding listeners and walking the file's parse tree.
+     *
+     * @param inputFile File to analyze.
+     * @param optTokenStream Common token stream for input file.
+     * @param optTree Parse tree for input file.
      * @throws CliArgumentParserException if an error occurs when parsing cmd line arguments
      */
-    public static SwiftParser.TopLevelContext getParseTree(CommonTokenStream tokenStream)
+    public static void analyzeFile(File inputFile, Optional<CommonTokenStream> optTokenStream,
+                                   Optional<TopLevelContext> optTree, Formatter formatter, Severity maxSeverity)
         throws CliArgumentParserException {
-        SwiftParser swiftParser = new SwiftParser(tokenStream);
-        if (!configuration.debugFlagSet()) {
-            swiftParser.removeErrorListeners();
-            swiftParser.addErrorListener(new ErrorListener());
+
+        ConstructLengths constructLengths = configuration.parseConstructLengths();
```

Comment:
Would it be more efficient to call these methods just once in the parent method and pass them in, rather than in the loop?